checks/mounts

```diff
@@ -29,17 +29,32 @@ def discovery_mounts(info):
         if dev not in devices:
             devices.append(dev)
             opts = sorted(options.split(","))
-            yield mountpoint.replace("\\040(deleted)", ""), opts
+            yield mountpoint.replace("\\040(deleted)", ""), {'expected': opts}
 
 
-def _should_ignore_option(option):
-    for ignored_option in ["commit=", "localalloc=", "subvol=", "subvolid="]:
+def _should_ignore_option(option, ignorelist):
+    for ignored_option in ignorelist:
         if option.startswith(ignored_option):
             return True
     return False
 
 
-def check_mounts(item, targetopts, info):
+def check_mounts(item, params, info):
+
+    # The original plugin had a parameter "targetopts" which was just a list
+    # of strings (expected mount options as determined during the discovery).
+    # Our modified plugin uses a dictionary as parameter with two elements:
+    #     'expected': same as the old "targetopts"
+    #     'ignore': list of strings (mount options to ignore)
+    # Both parameters can be overwritten using a WATO rule.
+    default_ignorelist = ["commit=", "localalloc=", "subvol=", "subvolid="]
+    if isinstance(params, dict):
+        targetopts = params.get('expected', [])
+        ignorelist = params.get('ignore', default_ignorelist)
+    else:
+        targetopts = params
+        ignorelist = default_ignorelist
+
```

Comment:
Theres a lot going on in this block :-)

1) The transform you implement below will be applied during cmk-update-config (to all stored rules, and to the already discovered services). You can assume `params` to be in the format your transform returns.
2) You should not need `params.get(key)`, but use `params[key]`. The 'expected' key must be present (because it is discovered), and the "ignored" key should be added as `factory_setting`. Then you can rely on both keys being present.